src/lib/Hydra/Controller/Build.pm

```diff
@@ -423,15 +421,15 @@ sub getDependencyGraph {
             };
         $$done{$path} = $node;
         my @refs;
-        foreach my $ref ($MACHINE_LOCAL_STORE->queryReferences($path)) {
+        foreach my $ref (binaryCacheStore()->queryReferences($path)) {
             next if $ref eq $path;
             next unless $runtime || $ref =~ /\.drv$/;
             getDependencyGraph($self, $c, $runtime, $done, $ref);
             push @refs, $ref;
         }
         # Show in reverse topological order to flatten the graph.
         # Should probably do a proper BFS.
-        my @sorted = reverse $MACHINE_LOCAL_STORE->topoSortPaths(@refs);
+        my @sorted = reverse binaryCacheStore()->topoSortPaths(@refs);
         $node->{refs} = [map { $$done{$_} } @sorted];
```

Comment:
Nit maybe `getDependencyGraph` should take a store parameter? Just for conceptual clarity?

Reply:
Agreed, done.